addons/sale_timesheet/models/product.py

```diff
@@ -24,15 +24,22 @@ def _selection_service_policy(self):
     service_upsell_threshold = fields.Float('Threshold', default=1, help="Percentage of time delivered compared to the prepaid amount that must be reached for the upselling opportunity activity to be triggered.")
     service_upsell_threshold_ratio = fields.Char(compute='_compute_service_upsell_threshold_ratio')
 
-    @api.depends('uom_id')
+    @api.depends('uom_id', 'company_id')
     def _compute_service_upsell_threshold_ratio(self):
         product_uom_hour = self.env.ref('uom.product_uom_hour')
+        uom_unit = self.env.ref('uom.product_uom_unit')
+        company_uom = self.env.company.timesheet_encode_uom_id
         for record in self:
-            if not record.uom_id or product_uom_hour.factor == record.uom_id.factor:
+            if not record.uom_id or product_uom_hour.factor == record.uom_id.factor or record.uom_id.category_id not in [product_uom_hour.category_id, uom_unit.category_id]:
```

Comment:
```suggestion
            if not record.uom_id or product_uom_hour.factor == record.uom_id.factor or\
               record.uom_id.category_id not in [product_uom_hour.category_id, uom_unit.category_id] or\
               record.uom_id != uom_unit:
```
Because the "else" block does the same thing as this "if" block.